crawler.py

```diff
@@ -97,54 +101,71 @@ def do_something(driver):
-            elem.send_keys('3')
-        elif input_type == 'submit':
+        if elem.tag_name in ['button', 'a']:
             elem.click()
-        elif input_type == 'color':
-            driver.execute_script("arguments[0].value = '#ff0000'", elem)
-        elif input_type == 'search':
-            elem.clear()
-            elem.send_keys('quick search')
-        elif input_type == 'radio':
-            elem.click()
-        else:
-            raise Exception('Unsupported input type: %s' % input_type)
-    elif elem.tag_name == 'select':
-        for option in elem.find_elements_by_tag_name('option'):
-            if option.text != '':
-                option.click()
-                break
-
-    close_all_windows_except_first(driver)
+        elif elem.tag_name == 'input':
+            input_type = elem.get_attribute('type')
+            if input_type == 'url':
+                elem.send_keys('http://www.mozilla.org/')
+            elif input_type == 'text':
+                elem.send_keys('marco')
+            elif input_type == 'email':
+                elem.send_keys('prova@email.it')
+            elif input_type == 'password':
+                elem.send_keys('aMildlyComplexPasswordIn2017')
+            elif input_type == 'checkbox':
+                elem.click()
+            elif input_type == 'number':
+                elem.send_keys('3')
+            elif input_type in ['submit', 'reset']:
+                elem.click()
+            elif input_type == 'color':
+                driver.execute_script("arguments[0].value = '#ff0000'", elem)
+            elif input_type == 'search':
+                elem.clear()
+                elem.send_keys('quick search')
+            elif input_type == 'radio':
+                elem.click()
+            else:
+                raise Exception('Unsupported input type: %s' % input_type)
+        elif elem.tag_name == 'select':
+            for option in elem.find_elements_by_tag_name('option'):
+                if option.text != '':
+                    option.click()
+                    break
+        elif elem.tag_name in ['dialog', 'frame']:
+            if len(elem.find_elements_by_tag_name('button')) != 0:
+                for button in elem.find_elements_by_tag_name('button'):
+                    button.click()
+                    break
+            elif len(elem.find_elements_by_tag_name('a')) != 0:
+                for link in elem.find_elements_by_tag_name('a'):
+                    if link.text != '':
+                        link.click()
+                        break
+        close_all_windows_except_first(driver)
+
+        return child_attributes
+
+    except (TimeoutException, ElementNotInteractableException, StaleElementReferenceException, WebDriverException):
+        # Ignore frequent exceptions.
+        traceback.print_exc()
+        not_clickable_elems.add(elem)
+        close_all_windows_except_first(driver)
 
-    return child_attributes
+    return 'Not clickable'
```

Comment:
If we clear the set at every iteration, we should return None here (as if we are here, it means we couldn't find any clickable element).